Student/StudentE2ETests/Dashboard/DashboardTests.swift

```diff
@@ -0,0 +1,226 @@
+        app.find(label: "No Courses").waitToExist()
+
+        // MARK: Create an enrollment and an announcement
+        let enrollment = seeder.enrollStudent(student, in: course, state: .invited)
+        let announcement = AnnouncementsHelper.postAccountNotification()
+        BaseHelper.pullToRefresh()
+
+        // MARK: Check visibility and order of the enrollment and the announcement
+        let courseAcceptButton = CourseInvitation.acceptButton(id: enrollment.id).waitToExist()
+        XCTAssertTrue(courseAcceptButton.isVisible)
+
+        let notificationToggleButton = AccountNotifications.toggleButton(id: announcement.id).waitToExist()
+        XCTAssertTrue(notificationToggleButton.isVisible)
+
+        XCTAssertLessThan(courseAcceptButton.frame().maxY, notificationToggleButton.frame().minY)
+    }
+
+    func testNavigateToDashboard() {
+        // MARK: Seed the usual stuff and a front page for the course
+        let student = seeder.createUser()
+        let course = seeder.createCourse()
+        seeder.enrollStudent(student, in: course)
+        DashboardHelper.createFrontPageForCourse(course: course)
+
+        // MARK: Get the user logged in and navigate to the course
+        logInDSUser(student)
+        var courseCard = Dashboard.courseCard(id: course.id).waitToExist()
+        XCTAssertTrue(courseCard.isVisible)
+        courseCard.tap()
+
+        // MARK: Navigate to pages of course and open front page
+        CourseNavigation.pages.tap()
+        PageList.frontPage.tap()
+
+        // MARK: Tap dashboard tab and check visibility of course card and label
+        TabBar.dashboardTab.tap()
+        let coursesLabel = Dashboard.coursesLabel.waitToExist()
+        courseCard = Dashboard.courseCard(id: course.id).waitToExist()
+        XCTAssertTrue(coursesLabel.isVisible)
+        XCTAssertTrue(courseCard.isVisible)
+    }
+
+    func testCourseCardInfo() {
+        // MARK: Seed the usual stuff
+        let student = seeder.createUser()
+        let course = seeder.createCourse()
+        seeder.enrollStudent(student, in: course)
+
+        // MARK: Get the user logged in and check visibility and label of course
+        logInDSUser(student)
+        let courseCard = Dashboard.courseCard(id: course.id).waitToExist()
+        XCTAssertTrue(courseCard.isVisible)
+        XCTAssertTrue(courseCard.label().contains(course.name))
+    }
+
+    func testDashboardEditButtonDisplaysCorrectCourses() {
+        // MARK: Seed the usual stuff with 7 courses and student enrolled in them with all 7 different states
+        let student = seeder.createUser()
+        let courses = DashboardHelper.createCourses(number: 7)
+        seeder.enrollStudent(student, in: courses[0], state: .active)
+        seeder.enrollStudent(student, in: courses[1], state: .invited)
+        seeder.enrollStudent(student, in: courses[2], state: .completed)
+        seeder.enrollStudent(student, in: courses[3], state: .creation_pending)
```

Comment:
Is there a way to test this one?

Reply:
These lines are inside a test case. And there are multiple courses here. Which course do you mean of these 4 that you highlighted? This particular test is for checking the visibility of every course on the edit screen of the dashboard, depending on the enrollment state of each. What step do you miss that should be tested here?